elixir/apps/web/lib/web/components/core_components.ex

```diff
@@ -1140,4 +786,36 @@ defmodule Web.CoreComponents do
   def translate_errors(errors, field) when is_list(errors) do
     for {^field, {msg, opts}} <- errors, do: translate_error({msg, opts})
   end
+
+  @doc """
+  Returns a string the represents a relative time for a given Datetime
+  from the current time or a given base time
+  """
+  attr :relative, DateTime, required: true
+  attr :relative_to, DateTime, required: false, default: DateTime.utc_now()
+
+  def relative_datetime(assigns) do
+    diff = DateTime.diff(assigns[:relative_to], assigns[:relative])
+
+    diff_str =
+      cond do
+        diff <= -24 * 3600 -> "in #{div(-diff, 24 * 3600)}day(s)"
+        diff <= -3600 -> "in #{div(-diff, 3600)}hour(s)"
+        diff <= -60 -> "in #{div(-diff, 60)}minute(s)"
+        diff <= -5 -> "in #{-diff}seconds"
+        diff <= 5 -> "now"
+        diff <= 60 -> "#{diff}seconds ago"
```

Comment:
Have you tried using https://github.com/elixir-cldr/cldr_dates_times for this? Specifically this module should have everything we need: https://github.com/elixir-cldr/cldr_dates_times/blob/main/lib/cldr/datetime/relative.ex

```suggestion
        diff <= -24 * 3600 -> "in #{div(-diff, 24 * 3600)} day(s)"
        diff <= -3600 -> "in #{div(-diff, 3600)}hour(s)"
        diff <= -60 -> "in #{div(-diff, 60)} minute(s)"
        diff <= -5 -> "in #{-diff} seconds"
        diff <= 5 -> "now"
        diff <= 60 -> "#{diff} seconds ago"
```

Reply:
Yeah, I was trying to, but the setup seemed to require some Elixir/OTP type knowledge that I felt like I was going down a rabbit hole with, so I figured I'd just encapsulate everything we need at the moment in this function and then we could refactor it easily later by pulling in the library you recommended and we could get rid of this all together.

Reply:
Please drop a comment then so that somebody who revisits the code will replace it instead of trying to fix it (if there are any bugs):
```elixir
# Note: This code will be replaced with https://github.com/elixir-cldr/cldr_dates_times/blob/main/lib/cldr/datetime/relative.ex in future
```

Reply:
Sounds good

Reply:
Comment has been added.